client/src/cmdlfem4x50.c

```diff
@@ -246,270 +121,501 @@ static void print_info_result(uint8_t *data, bool verbose) {
+    PacketResponseNG resp;
+    
+    if (!WaitForResponseTimeout(CMD_ACK, &resp, 8000)) {
+        PrintAndLogEx(WARNING, "Timeout while waiting for reply.");
+        return PM3_ETIMEOUT;
+    }
+    
+    isok = resp.oldarg[0] & 0xFF;
+    if (!isok) {
+        PrintAndLogEx(WARNING, "Flash error");
+        return PM3_EFLASH;
+    }
+    
+    return PM3_SUCCESS;
 }
 
-int CmdEM4x50Info(const char *Cmd) {
+static int em4x50_write_flash(uint8_t *data, int offset, size_t datalen) {
+    
+    int isok = 0;
+    uint32_t bytes_sent = 0;
+    uint32_t bytes_remaining = datalen;
+    uint32_t bytes_in_packet = 0;
+    PacketResponseNG resp;
 
-    // envoke reading of a EM4x50 tag which has to be on the antenna because
-    // decoding is done by the device (not on client side)
+    // wipe
+    em4x50_wipe_flash(0);
+    em4x50_wipe_flash(1);
+    em4x50_wipe_flash(2);
 
-    bool errors = false, verbose = false;
-    uint8_t cmdp = 0;
-    em4x50_data_t etd;
-    etd.pwd_given = false;
+    // fast push mode
+    conn.block_after_ACK = true;
 
-    while (param_getchar(Cmd, cmdp) != 0x00 && !errors) {
-        switch (tolower(param_getchar(Cmd, cmdp))) {
+    while (bytes_remaining > 0) {
+        bytes_in_packet = MIN(FLASH_MEM_BLOCK_SIZE, bytes_remaining);
 
-            case 'h':
-                return usage_lf_em4x50_info();
+        clearCommandBuffer();
+        SendCommandOLD(CMD_FLASHMEM_WRITE, offset + bytes_sent, bytes_in_packet, 0, data + bytes_sent, bytes_in_packet);
 
-            case 'p':
-                if (param_gethex(Cmd, cmdp + 1, etd.password, 8)) {
-                    PrintAndLogEx(FAILED, "\n  password has to be 8 hex symbols\n");
-                    return PM3_EINVARG;
-                }
-                etd.pwd_given = true;
-                cmdp += 2;
-                break;
+        bytes_remaining -= bytes_in_packet;
+        bytes_sent += bytes_in_packet;
 
-            case 'v':
-                verbose = true;
-                cmdp += 1;
-                break;
+        if (WaitForResponseTimeout(CMD_ACK, &resp, 2000) == false) {
+            PrintAndLogEx(WARNING, "timeout while waiting for reply.");
+            conn.block_after_ACK = false;
+            return PM3_ETIMEOUT;
+        }
 
-            default:
-                PrintAndLogEx(WARNING, "  Unknown parameter '%c'", param_getchar(Cmd, cmdp));
-                errors = true;
-                break;
+        isok = resp.oldarg[0] & 0xFF;
+        if (!isok) {
+            conn.block_after_ACK = false;
+            PrintAndLogEx(FAILED, "Flash write fail [offset %u]", bytes_sent);
+            return PM3_EFLASH;
         }
     }
 
-    // validation
-    if (errors)
-        return usage_lf_em4x50_info();
+    conn.block_after_ACK = false;
+    
+    return PM3_SUCCESS;
+}
 
-    clearCommandBuffer();
-    SendCommandNG(CMD_LF_EM4X50_INFO, (uint8_t *)&etd, sizeof(etd));
+int CmdEM4x50ELoad(const char *Cmd) {
```

Comment:
eload in the client is used for "emulator memory".  A specific part of Bigbuf ,  ram memory.
This seems to be a flash memory now,   which is specific for RDV4.  ... ...